pkg/fanal/artifact/artifact.go

```diff
@@ -86,23 +86,10 @@ type Artifact interface {
 	Clean(reference Reference) error
 }
 
-// Type represents a type of artifact
-type Type string
```

Comment:
`artifact` package imports many other modules, which cannot be compiled to Wasm. I moved this type to `fanal/types` so Wasm modules will not import `artifact`.